### PR TITLE
make the pauli_rep operator property public

### PR DESCRIPTION
**Pull request by timmysilv**
**Context:**
The `pauli_rep` property is widely used, and should be made public

**Description of the Change:**
- Introduce the `pauli_rep` read-only property to the `Operator` class
- Change all (read) references to `op._pauli_rep` into `op.pauli_rep` (the public version)

**Benefits:**
We're committing to having `pauli_rep` be part of the public API, making it less scary to lean on while moving forward with other improvements (eg. combining all the hamiltonian expansion transforms)

**Possible Drawbacks:**
We might want to change things related to it again, and it would require more careful work because it's public. I'm not so worried about this because it has existed (and been used) for a while now.

**Related GitHub Issues:**
previously attempted in #4017 
[sc-37494]

### CODE REVIEW

**Review by mudit2812 — APPROVED** [review]
Big fat thumbs up from me

File: pennylane/operation.py [hunk 1]
```diff
@@ -1231,6 +1231,11 @@ def hyperparameters(self):
         self._hyperparameters = {}
         return self._hyperparameters
 
+    @property
+    def pauli_rep(self):
+        """A PauliSentence representation of the Operator, or None if it doesn't have one."""
```
Comment (albi3ro) [inline pennylane/operation.py:1236]:
```suggestion
        """A :class:`~.PauliSentence` representation of the Operator, or ``None`` if it doesn't have one."""
```

**Comment by trbromley** [comment]
Thanks @timmysilv, no concerns from my side!

**Review by albi3ro — APPROVED** [review]
🎉

**Pull request merged**